src/spec/create_extension_spec.py

```diff
@@ -470,52 +426,99 @@ def main():
-                     "Uses trodes time (samplecount)."),
+                doc=(
+                    "If the ripple filter was used, lockout period in the number of samples. "
+                    "Uses trodes time (samplecount)."
+                ),
                 dtype="int",
                 quantity="?",
             ),
             NWBDatasetSpec(
                 name="ripple_filter_threshold_sd",
                 neurodata_type_inc="VectorData",
-                doc=("If the ripple filter was used, the threshold for detecting a ripple, in number of "
-                     "standard deviations."),
+                doc=(
+                    "If the ripple filter was used, the threshold for detecting a ripple, in number of "
+                    "standard deviations."
+                ),
                 dtype="float",
                 quantity="?",
             ),
             NWBDatasetSpec(
                 name="ripple_filter_num_above_threshold",
                 neurodata_type_inc="VectorData",
-                doc=("If the ripple filter was used, the number of tetrodes that have their signal cross "
-                     "the standard deviation threshold."),
+                doc=(
+                    "If the ripple filter was used, the number of tetrodes that have their signal cross "
+                    "the standard deviation threshold."
+                ),
                 dtype="int",
                 quantity="?",
             ),
+            NWBDatasetSpec(
+                name="speed_filter_on",
+                neurodata_type_inc="VectorData",
+                doc=(
+                    "Whether the speed filter was on. Closed-loop stimulation based on whether the animal "
+                    "is moving fast/slow enough. If this column is not present, then the speed filter was not used."
+                ),
+                dtype="bool",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="speed_filter_threshold",
+                neurodata_type_inc="VectorData",
+                doc=(
+                    "If the speed filter was used, the threshold for detecting a fast/slow animal, in cm/s."
+                ),
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="speed_filter_on_above_threshold",
+                neurodata_type_inc="VectorData",
+                doc=(
+                    "If the speed filter was used, True if active when speed above threshold."
+                ),
+                dtype="bool",
+            ),
+            NWBAttributeSpec(
+                name="stimulus_signal",
+                neurodata_type_inc="VectorData",
+                doc=(
+                    "Timeseries of the delivered stimulus. Can be continuous values or time of digital on/off events."
+                ),
+                dtype="object",
```

Comment:
```suggestion
                dtype=NWBRefSpec(
                    target_type="TimeSeries",
                    reftype="object",
                ),
```

`dtype="object"` is not a valid option. I'm surprised this did not raise an error - I'll look into that. To create an object reference to a TimeSeries, you'll have to do the above.